apis/v1alpha1/cluster_tc_program_types.go

```diff
@@ -20,39 +20,60 @@ package v1alpha1
 // +kubebuilder:validation:Enum:=UnSpec;OK;ReClassify;Shot;Pipe;Stolen;Queued;Repeat;ReDirect;Trap;DispatcherReturn;
 type TcProceedOnValue string
 
-// ClTcProgramInfo defines the tc program details
 type ClTcProgramInfo struct {
-	// The list of points to which the program should be attached.  The list items
-	// are optional and may be udated after the bpf program has been loaded
+	// links is optional and is the list of hook points to which the TC program
+	// should be attached. The TC program is loaded in kernel memory when the BPF
+	// Application CRD is created and the selected Kubernetes nodes are active.
+	// The TC program will not be triggered until the program has also been
+	// attached to an attachment point described in this list. Items may be added
+	// or removed from the list at any point, causing the TC program to be
+	// attached or detached.
+	//
+	// The hook point for a TC program is an interface (or network device). An
+	// interface can be specified by name, or using the primaryNodeInterface flag,
+	// which indicates to use the primary interface of a Kubernetes node or
+	// container. Optionally, the TC program can be installed in a set of
+	// containers.
```

Comment:
The hook point for a TC program is a network interface (or device). The
interface can be specified by name, or by setting the primaryNodeInterface
flag, which instructs bpfman to use the primary interface of a Kubernetes
node. Optionally, the TC program can also be installed into a
set of network namespaces.

Reply:
Just a slight rewording suggestion.  Also, containers will change to network namespaces after a rebase.

This comment also applies to TCX and XDP.

Reply:
Also, primaryNodeInterface doesn't work inside containers.  It currently always finds the primary node interface of the node.  Coincidentally, on kind this seems to be `eth0` for both the node and all the pods.